frontend/src/components/elements/upload/IngredientItems.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react";
+import { DigitText, DigitTooltip } from "@cthit/react-digit-components";
+import "./styles/IngredientItems.css";
+
+class IngredientItems extends Component {
+  handleDelete = ingredient => {
+    this.props.handleDelete(ingredient);
+  };
+
+  formatText = ingredient => {
+    let ing = ingredient[0];
+    let amount = ingredient[1];
+    let meassurement = ingredient[2];
+
+    return amount + " " + meassurement + " " + ing;
+  };
+
+  render() {
+    return (
+      <ul className="ingredientList">
+        {this.props.ingredients.map(ingredient => {
+          let itemKey = Math.floor(Math.random() * 1000 + 1);
```

Comment:
Don't use Math.random when generating a key. Use instead something in ingredient, a prop or id, that's unique. A new key will be generated on each render, making the usage of key here useless